Sources/Core/UInt.swift

```diff
@@ -9,6 +9,23 @@
 public struct UInt {
   @usableFromInline
   internal var _value: Builtin.Int32
+
+  @_transparent
+  public init(_ _value: Builtin.Int32) {
+    self._value = _value
+  }
+
+  @_transparent
+  public static func &= (lhs: inout UInt, rhs: UInt) {
+    lhs = UInt(Builtin.and_Int32(lhs._value, rhs._value))
+  }
+
+  @_transparent
+  public static func & (lhs: UInt, rhs: UInt) -> UInt {
+    var lhs = lhs
+    lhs &= rhs
+    return lhs
+  }
```

Comment:
Hmm, can we flip the implementation around?

```
public static func & (_ lhs: UInt, _ rhs: UInt) -> UInt {
  return UInt(Builtin.and_Int32(lhs._value, rhs._value))
}

public static func &= (_ lhs: inout UInt, _ rhs: UInt) {
  lhs = lhs & rhs
}
```

Reply:
The way it currently is in my PR is how "real" stdlib implemented it. I don't think that implementation had a clarifying comment, but I just assumed they had a good (maybe slight performance difference?) reason for it.

Reply:
Well, simply copying the reference implementation isn't particularly a goal :).

I'm open to the idea that we need to write this this way, but I would rather have some reason to do that.  I would rather that the implementation be clear and concise with good reason for the complexity.

I suspect that this will end up the same once the optimizer has run through - the `@_transparent` should fully inline the operation as if it were written there.  However, if I am incorrect and there is a size implication, then I would prefer the variant that has the smaller output (code) size.

Reply:
Are you referring to the potential code size difference in uSwift clients? If so, how should we measure that? Do we need some kind of test suite for code size testing?

Reply:
Yes, we really do need a way to track the code-size impact of changes.  I'd like to use bloaty for that, but I would be okay with anything which tracks the size over time.  I assume that for most embedded cases, we would want static linking, but, the size of the library either static or dynamic is important to consider.

Reply:
BTW, the actual stdlib does this for all numeric operators too: they use `Builtin` within assignment operators, and non-assignment operators are implemented in terms of those.

Reply:
Thats certainly good to know.  However, being 1:1 identical to the default standard library is a non-goal.  The desire is to make it easy to use as broad of a surface in space constrained environments as possible.  As a more concrete example, I don't expect `String` to be fully compatible because not all environments need to have all the various string encodings.  If we find that we can do anything to reduce code size (or improve clarity) I would favour those.